lib/erl_docgen/src/eep48_to_markdown.erl

```diff
@@ -144,19 +147,52 @@ convert(File, Docs) ->
           fun(MFA1, MFA2) ->
                   erl_anno:line(element(2, MFA1)) >= erl_anno:line(element(2, MFA2))
           end, Docs),
-    convert(string:split(File, "\n", all), [], SortedDocs).
-convert(File, Acc, []) ->
-    unicode:characters_to_binary([[A,$\n] || A <- File ++ Acc]);
-convert(File, Acc, [{_, Anno, _, #{ <<"en">> := D }, _} | T]) ->
-    {Before, After} = lists:split(erl_anno:line(Anno)-1, File),
-    convert(Before, [comment(render_docs(D, init_config(D, #{})))|After] ++ Acc, T).
+    Lines = string:split(File, "\n", all),
+    {ModuleHeader, ModuleBody} =
+        lists:foldl(fun document_module/2, {Lines, []}, SortedDocs),
+    ModuleHeader ++ ModuleBody.
 
+document_module({_, Anno, _, #{ <<"en">> := D }, _}, {File, Acc}) ->
+    {Before, After} = lists:split(erl_anno:line(Anno)-1, File),
+    DocModule = comment(render_docs(D, init_config(D, #{}))),
+    {Before, [DocModule|After] ++ Acc}.
+
+%% Convert module documentation
+convert_moduledoc(ModuleHeader) ->
+    DocHeader = lists:flatmap(
+                  fun (Doc) ->
+                          render_docs(Doc, init_config(Doc, #{}))
+                  end, ModuleHeader),
+    moduledoc(DocHeader).
+
+format_doc(FileModuleDoc, FileFunDoc) ->
+    {Copyright, Module} = split_by_copyright(FileFunDoc),
+    ModuleWithDocs = Copyright ++ FileModuleDoc ++ Module,
+    unicode:characters_to_binary([[A,$\n] || A <- ModuleWithDocs]).
+
+split_by_copyright(File) ->
+    split_by_copyright(File, []).
+split_by_copyright([<<"-module", _/binary>> | _]=Module, Acc) ->
+    {lists:reverse(Acc), Module};
+split_by_copyright([B | Bs], Acc) when is_binary(B); is_list(B) ->
+    split_by_copyright(Bs, [B | Acc]).
+
+
+%% Comments are context-dependent:
+%% - a comment in the doc chunk cannot be considered a document attribute unless it relates to a function
+%% - a comment at the top level is a -moduledoc attribute
+%% - future work: doc strings if there are examples in code
 comment(String) ->
-    ["-doc \"",
+    ["-doc \"\"\"\n",
```

Comment:
I used `"` instead of `"""` so that the current compiler can compile the modules after we have converted them.